solidity/contracts/converter/types/standard-pool/StandardPoolConverter.sol

```diff
@@ -0,0 +1,1386 @@
+        require(reserveId != 0, "ERR_INVALID_RESERVE");
+        return reserveBalance(reserveId);
+    }
+
+    /**
+     * @dev returns the balances of both reserve tokens
+     *
+     * @return the balances of both reserve tokens
+     */
+    function reserveBalances() public view returns (uint256, uint256) {
+        return reserveBalances(1, 2);
+    }
+
+    /**
+     * @dev converts a specific amount of source tokens to target tokens
+     * can only be called by the bancor network contract
+     *
+     * @param _sourceToken source ERC20 token
+     * @param _targetToken target ERC20 token
+     * @param _amount      amount of tokens to convert (in units of the source token)
+     * @param _trader      address of the caller who executed the conversion
+     * @param _beneficiary wallet to receive the conversion result
+     *
+     * @return amount of tokens received (in units of the target token)
+     */
+    function convert(
+        IERC20Token _sourceToken,
+        IERC20Token _targetToken,
+        uint256 _amount,
+        address _trader,
+        address payable _beneficiary
+    ) public payable override protected only(BANCOR_NETWORK) returns (uint256) {
+        // validate input
+        require(_sourceToken != _targetToken, "ERR_SAME_SOURCE_TARGET");
+
+        return doConvert(_sourceToken, _targetToken, _amount, _trader, _beneficiary);
+    }
+
+    /**
+     * @dev returns the conversion fee for a given target amount
+     *
+     * @param _targetAmount  target amount
+     *
+     * @return conversion fee
+     */
+    function calculateFee(uint256 _targetAmount) internal view returns (uint256) {
+        return _targetAmount.mul(conversionFee) / PPM_RESOLUTION;
+    }
+
+    /**
+     * @dev loads the stored reserve balance for a given reserve id
+     *
+     * @param _reserveId   reserve id
+     */
+    function reserveBalance(uint256 _reserveId) internal view returns (uint256) {
+        return decodeReserveBalance(__reserveBalances, _reserveId);
+    }
+
+    /**
+     * @dev loads the stored reserve balances
+     *
+     * @param _sourceId    source reserve id
+     * @param _targetId    target reserve id
+     */
+    function reserveBalances(uint256 _sourceId, uint256 _targetId) internal view returns (uint256, uint256) {
+        require((_sourceId == 1 && _targetId == 2) || (_sourceId == 2 && _targetId == 1), "ERR_INVALID_RESERVES");
+        return decodeReserveBalances(__reserveBalances, _sourceId, _targetId);
+    }
+
+    /**
+     * @dev stores the stored reserve balance for a given reserve id
+     *
+     * @param _reserveId       reserve id
+     * @param _reserveBalance  reserve balance
+     */
+    function setReserveBalance(uint256 _reserveId, uint256 _reserveBalance) internal {
+        require(_reserveBalance <= MAX_UINT128, "ERR_RESERVE_BALANCE_OVERFLOW");
+        uint256 otherBalance = decodeReserveBalance(__reserveBalances, 3 - _reserveId);
```

Comment:
re: `3 - ...` - can you refactor all these reserves IDs magic numbers (1, 2, and 3) to some helper functions? Otherwise, it's hard to read and error prone

Reply:
Yes. 3 is their sum of course, but you're right.

Reply:
We later use `(_id - 1) * 128` which relies on the fact that this ID can be either `1` or `2`.
Any suggestion on how to generalize this?
Just declaring constants `1` and `2` implies that one can change them to any other values.

Reply:
I see. Let's leave it for now